### ISSUE TITLE
Safety/correctness checker for OpenSearch software

### ISSUE THREAD

**Issue by Bukhtawar** [opensearch-project/OpenSearch#3866]
**Is your feature request related to a problem? Please describe.**
With deep changes being made to core espl in the Engine with the new [segment based replication strategy](https://github.com/opensearch-project/OpenSearch/issues/2229) and it's integration with [remote segment store and translog](https://github.com/opensearch-project/OpenSearch/issues/1968), it becomes imperative that we re-evaluate safety, correctness and data integrity properties of the system in the face of common system failure modes. 

**Describe the solution you'd like**
Mechanism to inject failures and verify system behaviour for correctness, safety and data integrity

**Describe alternatives you've considered**
[Jepsen](https://github.com/opensearch-project/OpenSearch/issues/1968) is one of the various softwares(not AL2.0) that provides similar capabilities.

**Additional context**

**Comment by Bukhtawar** [comment]
cc: @mch2 @sachinpkale @ashking94 @kartg @andrross @reta @nknize 
Looking for thoughts on adding this framework.

**Comment by reta** [comment]
+1 to look into [Jepsen](https://github.com/jepsen-io/jepsen), Elastic has formal TLA+ models published (AL 2.0) [1] which could be quite helpful, although no updates since 2019.

[1] https://github.com/elastic/elasticsearch-formal-models

**Comment by andrross** [comment]
I think this is great idea @Bukhtawar. What do you think the best way to get started is? We could start from the TLA+ models referenced above and modify as necessary (if anything is out of date from 2019). Or we could start fresh and build on top of that.

**Comment by sachinpkale** [comment]
I completely agree with the proposal to add correctness checker. We started listing down failure scenarios of remote segment store and realized that there are many corner cases and almost impossible to cover them manually.

**Comment by Bukhtawar** [comment]
@andrross I guess we could start from the existing models that are there and see if they need an update or need more comprehensive cases for remote store and recent engine changes that we are making
